.github/workflows/python-code-analysis.yml

```
@@ -0,0 +1,25 @@
+name: Python code analysis
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - '.github/workflows/python-code-analysis.yml'
+  schedule:
+    - cron: '0 10 * * 3'
```

Have it run on Tuesdays instead:
```suggestion
    - cron: '0 10 * * 2'
```